__tests__/defaultProvider.test.ts

```diff
@@ -1,11 +1,11 @@
-import { BlockNumber, GetBlockResponse, stark } from '../src';
+import { BlockNumber, GetBlockResponse, Provider, stark } from '../src';
 import { toBN } from '../src/utils/number';
 import { encodeShortString } from '../src/utils/shortString';
 import { compiledErc20, erc20ClassHash, getTestAccount, getTestProvider } from './fixtures';
 
 const { compileCalldata } = stark;
 
-const testProvider = getTestProvider();
+const testProvider = new Provider(getTestProvider());
```

Comment:
This translates to
`new Provider(new Provider(..))`
Why do we need it like this ?

Reply:
It actually doesn't, `getTestProvider` returns either an `RpcProvider` or `SequencerProvider`, not a `Provider` instance so it equates either to `new Provider(new RpcProvider(..))` or `new Provider(new SequencerProvider(..))`. The tests work since they all conform to the `ProviderInterface`, however, bugs are still possible, e.g. when `getClassHashAt` had flipped parameters between the specific providers and a mapping in the default provider.

We spoke about changing the default provider implementation to not be a wrapping proxy for the specific providers, but until that happens this is the more precise setup for the default provider tests.

Reply:
Yea, Good point!